### PR TITLE
feat(deps): Migrate to apollo v4 [BACK-1736]

### PR DESCRIPTION
**Pull request by hyperparabolic**
## Goal
Migrate to apollo v4

- Refactor server startup to be in an async scope, and initialize full express server in startServer
- Explicitly check for 'true' value in premium header in ContextManager.
- Integration tests have all been refactored to execute through the express server using supertest.

## I'd love feedback/perspectives on:
- Is it safe to check for the string `'true'` in `request.headers.premium`? Signs point to yes based on the [client-api docs](https://studio.apollographql.com/graph/pocket-client-api/home?variant=current), but I'd like a double check. The previous implementation evaluates 'false' as a true value, and that just doesn't seem right.
- Is it normal for the sqs queue to not get set up in localstack when running integration tests locally? The tests pass, but there are a ton of errors being dumped in the console because the sqs queue doesn't exist. Hoping this isn't the case in circleci. Might need to do some more debugging if those logs show up in circleci as well.

## Deployment steps
N/A

## References

JIRA ticket:
https://getpocket.atlassian.net/browse/BACK-1736?atlOrigin=eyJpIjoiNjA3NDE0ZTRhMTIwNDYyNGI0ZTEyZTRlNTIzYjc1NGMiLCJwIjoiaiJ9

[BACK-1736]: https://getpocket.atlassian.net/browse/BACK-1736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hyperparabolic** [comment]
Hrm..  Two things to debug here:

- CodeBuild is failing during `tfenv install` as of 6 days ago?  `curl: (48) An unknown option was passed in to libcurl`
- The SQS queue is not being set up in circleci either.  I feel like this might be related to `BatchDeleteHandler` lifecycle management? Digging at this now.

File: src/main.ts
Comment (hyperparabolic) [inline src/main.ts:27]: As far as I can tell, SQS is not set up in localstack, and there's nothing to poll while running integration tests.  Is this true?

**Comment by hyperparabolic** [comment]
Codebuild issues are resolved here with the base changed.  Do not merge this until that is merged into main though.

**Comment by hyperparabolic** [comment]
All of the CI blockers are merged into main, and I've double checked source of the premium header in [client-api](https://github.com/Pocket/client-api/blob/818bff40499f170ea14a0dfb33a4129b8e4f1fa8/src/jwtUtils.ts#L18), and it is confirmed a boolean before casting to a string, so it should be "true" or "false" reaching this service.  Other than needing a review, I think this is good to go.

**Review by cmharlow — DISMISSED** [review]

**Pull request merged**
